text/0000-build-systems.md

```diff
@@ -0,0 +1,446 @@
+- Feature Name: N/A
+- Start Date: 2017-09-01
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This **experimental RFC** lays out a high-level plan for improving Cargo's
+ability to integrate with other build systems and environments. As an
+experimental RFC, it opens the door to landing [unstable features] in Cargo to
+try out ideas, but *not* to stabilizing those features, which will require
+follow-up RFCs. It proposes a variety of features which, in total, permit a wide
+spectrum of integration cases -- from customizing a single aspect of Cargo to
+letting an external build system run almost the entire show.
+
+[unstable features]: https://github.com/rust-lang/cargo/pull/4433/
+
+# Motivation
+[motivation]: #motivation
+
+One of the first hurdles for using Rust in production is integrating it into
+your organization's build system. The level of challenge depends on the level of
+integration required: it's relatively painless to invoke Cargo from a makefile
+and let it fully manage building Rust code, but gets harder as you want the
+external build system to exert finer-grained control over how Rust code is built.
+The goal of this RFC is to lay out a vision for making integration at *any* scale
+much easier than it is today.
+
+After extensive discussion with stakeholders, there appear to be two distinct
+kinds of use-cases (or "customers") involved here:
+
+- **Mixed build systems**, where building already involves a variety of
+  language- or proeject-specific build systems. For this use case, the desire is
+  to use Cargo as-is, except for some specific concerns. Those concerns take a
+  variety of shapes: customizing caching, having a local crate registry, custom
+  handling for native dependencies, and so on. Addressing these concerns well
+  means adding new points of extensibility or control to Cargo.
+
+- **Homogenous build systems** like [Bazel], where there is a single prevailing
```

Comment:
homogeneous

Reply:
Can be written both ways apparently https://en.oxforddictionaries.com/definition/us/homogeneous

Reply:
That citation clearly indicates that homogeneous is still far more popular (“and around a third of citations for the word now use the form homogenous”). Homogenous was an error that has only gained *any* legitimacy because so many people make it! (As I like to say: language is a popularity contest.)